osfoffline/views/start_screen.py

```diff
@@ -21,47 +31,68 @@ def __init__(self):
-        # assumption: all the fields needed for db are set
-        full_name = user_name
-        osf_id = user_name
-        oauth_token = password
-        # check if user already exists in db
-
+        logging.debug('attempting to log in')
+        personal_access_token = self.start_screen.personalAccessTokenEdit.text().strip()
+        url = settings.API_BASE + '/v2/users/me/'
+        loop = asyncio.get_event_loop()
+        osf = OSFQuery(loop, personal_access_token)
+        logging.debug('url: {}'.format(url))
         try:
-            user = session.query(User).filter(User.osf_login == user_name).one()
-            user.logged_in = True
-            save(session, user)
-            self.close()
-        except MultipleResultsFound:
-            logging.warning(
-                'multiple users with same username. deleting all users with this username. restarting function.')
-            for user in session.query(User):
-                if user.osf_login == user_name:
-                    session.delete(user)
-                    save(session)
-            self.log_in()
-        except NoResultFound:
-            logging.warning('user doesnt exist. Creating user. and logging them in.')
-            user = User(
-                full_name=full_name,
-                osf_id=osf_id,
-                osf_login=user_name,
-                osf_local_folder_path='',
-                oauth_token=oauth_token,
-                osf_password=password
+            resp = yield from osf.make_request(url)
+        except (aiohttp.errors.ClientTimeoutError, aiohttp.errors.ClientConnectionError, concurrent.futures._base.TimeoutError):
+            # No internet connection
+            QMessageBox.warning(
+                None,
+                "Log in Failed",
+                "Unable to connect to server. Check your internet connection or try again later."
+            )
+        except aiohttp.errors.HttpBadRequest:
+            # Invalid credentials
+            QMessageBox.warning(
+                None,
+                "Log in Failed",
+                "Invalid login credentials."
             )
-            user.logged_in = True
-            save(session, user)
+        else:
+            if resp.status == 200:
+                json_resp = yield from resp.json()
+                full_name = json_resp['data']['attributes']['full_name']     # Use one of these later to "Welcome {user}"
+                given_name = json_resp['data']['attributes']['given_name']   # Use one of these later to "Welcome {user}"
+                osf_id = json_resp['data']['id']
+
+                try:
+                    user = session.query(User).filter(User.osf_id == osf_id).one()
+                    user.logged_in = True
+                    save(session, user)
+                except MultipleResultsFound:
+                    logging.warning('multiple users with same username. deleting all users with this username. restarting function.')
```

Comment:
Good point on the improvement ticket. We have some everyday user-facing bugs to triage first, and then we can tackle edge cases that will *create* bugs later.